scanpullrequest/scanpullrequest.go

```diff
@@ -150,23 +164,25 @@ func auditPullRequest(repoConfig *utils.Repository, client vcsclient.VcsClient)
 
 	defer func() {
 		if issuesCollection != nil {
-			xsc.SendScanEndedEvent(scanDetails.XrayVersion, scanDetails.XscVersion, scanDetails.ServerDetails, scanDetails.MultiScanId, scanDetails.StartTime, issuesCollection.CountIssuesCollectionFindings(), err)
+			xsc.SendScanEndedEvent(scanDetails.XrayVersion, scanDetails.XscVersion, scanDetails.ServerDetails, scanDetails.MultiScanId, scanDetails.StartTime, issuesCollection.GetTotalIssues(true), err)
 		}
 	}()
 
-	issuesCollection = &utils.IssuesCollection{}
+	issuesCollection = &issues.ScansIssuesCollection{}
 	for i := range repoConfig.Projects {
 		scanDetails.SetProject(&repoConfig.Projects[i])
-		var projectIssues *utils.IssuesCollection
+		var projectIssues *issues.ScansIssuesCollection
 		if projectIssues, err = auditPullRequestInProject(repoConfig, scanDetails); err != nil {
+			// Make sure status on scans are passed to show in the summary
+			issuesCollection.AppendStatus(projectIssues.ScanStatus)
```

Comment:
There is some issue here that I think might cause a panic. 
In ScansIssuesCollection.Append we first check if issues == nil (which means it can be nil, I assume if all the scans has failed). In this case addressing projectIssues.ScanStatus will cause a nil pointer panic.
In addition we append the statuses twice - here and in ScansIssuesCollection.Append (duplications).
if the issues != nil we always append them twice always and if issues == nil we get a panic